tools/astarte_export/lib/fetchdata/fetchdata.ex

```diff
@@ -0,0 +1,398 @@
+          map[:reception_timestamp]
+          |> DateTime.from_unix!(:millisecond)
+          |> DateTime.to_iso8601()
+
+        list = Map.to_list(map)
+
+        value_list =
+          List.foldl(list, [], fn {key, value}, acc ->
+            case to_string(key) do
+              "v_" <> item ->
+                match_object =
+                  Enum.find(extract_2nd_level_params, fn map1 -> map1[:suffix_path] == item end)
+
+                case match_object do
+                  nil ->
+                    acc
+
+                  _ ->
+                    data_type = match_object[:data_type]
+                    token = "/" <> match_object[:suffix_path]
+                    value1 = from_native_type(value, :double)
+
+                    case value1 do
+                      "" ->
+                        acc
+
+                      _ ->
+                        [%{:name => token, :value => value1} | acc]
+                    end
+                end
+
+              _Other ->
+                acc
+            end
+          end)
+
+        Map.new()
+        |> Map.put(:reception_timestamp, reception_timestamp)
+        |> Map.put(:path, path)
+        |> Map.put(:value, value_list)
+      end)
+  end
+
+  defp fetch_individual_properties(conn, realm, mappings, device_id, interface_id) do
+    Enum.reduce(mappings, [], fn mapping, acc1 ->
+      endpoint_id = mapping.endpoint_id
+      path = mapping.endpoint
+      data_type = mapping.value_type
+      data_field = get_data_field_name(data_type)
+
+      {:ok, result} =
+        Queries.retrive_individual_properties(conn, realm, device_id, interface_id, data_field)
+
+      values =
+        Enum.to_list(result)
+        |> Enum.map(fn map ->
+          reception_timestamp =
+            map[:reception_timestamp]
+            |> DateTime.from_unix!(:millisecond)
+            |> DateTime.to_iso8601()
+
+          path = map[:path] |> Kernel.to_string()
+
+          atom_data_field = String.to_atom(data_field)
+          return_value = map[atom_data_field]
+          value = from_native_type(return_value, data_type)
+
+          Map.new()
+          |> Map.put(:reception_timestamp, reception_timestamp)
+          |> Map.put(:path, path)
+          |> Map.put(:value, value)
+        end)
+    end)
+  end
+
+  def get_data_field_name(data_type) when is_atom(data_type) do
+    case data_type do
+      :double -> "double_value"
+      :integer -> "integer_value"
+      :boolean -> "boolean_value"
+      :longinteger -> "longinteger_value"
+      :string -> "string_value"
+      :binaryblob -> "binaryblob_value"
+      :datetime -> "datetime_value"
+      :doublearray -> "doublearray_value"
+      :integerarray -> "integerarray_value"
+      :booleanarray -> "booleanarray_value"
+      :longintegerarray -> "longintegerarray_value"
+      :stringarray -> "stringarray_value"
+      :binaryblobarray -> "binaryblobarray_value"
+      :datetimearray -> "datetimearray_value"
+      _ -> :error
```

Comment:
Here you're returning `:error` but the calling code never checks the return value and it ends up reaching the database. You should wrap correct results with `{:ok, result}` so that the caller can match in a `with` or a `case`